### PR TITLE
libsoxr: Fix pkgconfig file to relocate path

### LINKED ISSUE #20042: mingw-w64-libsoxr has broken pkgconf file
### Description / Steps to reproduce the issue

The installed soxr.pc file (and soxr-lsr.pc) ...

```
Name: soxr
Description: High quality, one-dimensional sample-rate conversion library
Version: 0.1.3
Libs: -L/ucrt64/lib -lsoxr
Cflags: -I/ucrt64/include
```

...has hardcoded paths and does not use the ${prefix} substitution that allows it to be successfully used in CMake.  i.e.

```
prefix=/ucrt64
exec_prefix=${prefix}
libdir=${prefix}/lib
includedir=${prefix}/include

Name: soxr
Description: High quality, one-dimensional sample-rate conversion library
Version: 0.1.3
Libs: -L${libdir} -lsoxr
Cflags: -I${includedir}
```

### Expected behavior

successful cmake inclusion using PkgConfig package.

### Actual behavior

```
CMake Error in CMakeLists.txt:
  Imported target "PkgConfig::LIBDEPS" includes non-existent path

    "/ucrt64/include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
```

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MINGW64_NT-10.0-19045 IF005-LT 3.4.10.x86_64 2024-02-10 08:39 UTC x86_64 Msys

### MINGW environments affected

- [ ] MINGW64
- [ ] MINGW32
- [X] UCRT64
- [ ] CLANG64
- [ ] CLANG32
- [ ] CLANGARM64

### Are you willing to submit a PR?

_No response_

### PR DESCRIPTION
**Pull request by Biswa96**
Fixes #20042

### CODE REVIEW

File: mingw-w64-libsoxr/0001-libsoxr-fix-pkgconfig-file.patch [hunk 1]
```diff
@@ -0,0 +1,68 @@
+diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
+index bb01a0d..7738867 100644
+--- a/src/CMakeLists.txt
++++ b/src/CMakeLists.txt
+@@ -86,9 +86,9 @@ set_target_properties (${PROJECT_NAME} PROPERTIES
+   PUBLIC_HEADER "${PROJECT_NAME}.h")
+ if (BUILD_FRAMEWORK)
+   set_target_properties (${PROJECT_NAME} PROPERTIES FRAMEWORK TRUE)
+-elseif (NOT WIN32)
++elseif (WIN32)
```
Comment (MehdiChinoune) [inline mingw-w64-libsoxr/0001-libsoxr-fix-pkgconfig-file.patch:10]:
```suggestion
-elseif (NOT WIN32)
+else()
```

File: mingw-w64-libsoxr/0001-libsoxr-fix-pkgconfig-file.patch [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
++  configure_file (${CMAKE_CURRENT_SOURCE_DIR}/${PROJECT_NAME}.pc.in ${TARGET_PCS} @ONLY)
+   install (FILES ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig)
+ endif ()
+ 
+@@ -110,9 +110,9 @@ if (WITH_LSR_BINDINGS)
+     PUBLIC_HEADER "${LSR}.h")
+   if (BUILD_FRAMEWORK)
+     set_target_properties (${LSR} PROPERTIES FRAMEWORK TRUE)
+-  elseif (NOT WIN32)
++  elseif (WIN32)
```
Comment (MehdiChinoune) [inline mingw-w64-libsoxr/0001-libsoxr-fix-pkgconfig-file.patch:22]:
```suggestion
-  elseif (NOT WIN32)
+  else()
```

File: mingw-w64-libsoxr/0001-libsoxr-fix-pkgconfig-file.patch (hunk 1 above)
Comment (Biswa96) [reply mingw-w64-libsoxr/0001-libsoxr-fix-pkgconfig-file.patch:10]: The effect is same though.

Comment (MehdiChinoune) [reply mingw-w64-libsoxr/0001-libsoxr-fix-pkgconfig-file.patch:10]: but, yours is not upstreamable.

Comment (Biswa96) [reply mingw-w64-libsoxr/0001-libsoxr-fix-pkgconfig-file.patch]: upstream is dead.

**Pull request merged**